packages/dcos-integration-test/extra/test_applications.py

```diff
@@ -109,85 +117,81 @@ def test_if_marathon_pods_can_be_deployed_with_mesos_containerizer(dcos_api_sess
-            },
-            'gracePeriodSeconds': 5,
-            'intervalSeconds': 10,
-            'timeoutSeconds': 10,
-            'maxConsecutiveFailures': 3
-        }]
-    }
-
-    dcos_api_session.marathon.deploy_and_cleanup(app_definition)
+def test_octarine(dcos_api_session, timeout=30):
+    # This app binds to port 80. This is only required by the http (not srv)
+    # transparent mode test. In transparent mode, we use ".mydcos.directory"
+    # to go to localhost, the port attached there is only used to
+    # determine which port to send traffic to on localhost. When it
+    # reaches the proxy, the port is not used, and a request is made
+    # to port 80.
+
+    octarine = "/opt/mesosphere/bin/octarine"
+    app, uuid = get_test_app()
+    app['acceptedResourceRoles'] = ["slave_public"]
+    app['portDefinitions'][0]["port"] = 80
+    app['requirePorts'] = True
+
+    with dcos_api_session.marathon.deploy_and_cleanup(app) as service_points:
+        port_number = service_points[0].port
+        # It didn't actually grab port 80 when requirePorts was unset
+        assert port_number == app['portDefinitions'][0]["port"]
+
+        app_name = app["id"].strip("/")
+        port_name = app['portDefinitions'][0]["name"]
+        port_protocol = app['portDefinitions'][0]["protocol"]
+
+        srv = "_{}._{}._{}.marathon.mesos".format(port_name, app_name, port_protocol)
+        addr = "{}.marathon.mesos".format(app_name)
+        transparent_suffix = ".mydcos.directory"
+
+        standard_mode = "standard"
+        transparent_mode = "transparent"
+
+        transparent_srv_bind = 2509
+
+        standard_addr = "{}:{}/ping".format(addr, port_number)
+        standard_srv = "{}/ping".format(srv)
+        transparent_addr = "{}{}:{}/ping".format(addr, transparent_suffix, transparent_srv_bind)
+        transparent_srv = "{}{}:{}/ping".format(srv, transparent_suffix, transparent_srv_bind)
+
+        # The uuids are different between runs so that they don't have a
+        # chance of colliding. They shouldn't anyways, but just to be safe.
+        octarine_runner(octarine, standard_mode, uuid + "1", standard_addr)
+        octarine_runner(octarine, standard_mode, uuid + "2", standard_srv)
+        octarine_runner(octarine, transparent_mode, uuid + "3", transparent_addr, bind_port=transparent_srv_bind)
+        octarine_runner(octarine, transparent_mode, uuid + "4", transparent_srv, bind_port=transparent_srv_bind)
+
+
+def octarine_runner(octarine, mode, uuid, uri, bind_port=None):
+    log.info("Running octarine(mode={}, uuid={}, uri={}".format(mode, uuid, uri))
+
+    bind_port_str = ""
+    if bind_port is not None:
+        bind_port_str = "-bindPort {}".format(bind_port)
+
+    server_cmd = "{} -mode {} {} {}".format(octarine, mode, bind_port_str, uuid)
+    log.info("Server: {}".format(server_cmd))
+    server = threading.Thread(target=subprocess.call, args=(shlex.split(server_cmd),), daemon=True)
```

Comment:
Why are we running this directly on the host instead of a mesos container? And why are we using a thread to control a subprocess, that seems redundant any python threads tend to introduce more bugs. E.G. you use daemon mode so if this thread hangs it will just be cleaned up with the the entire interpreter, but because that thread is starting a subprocess, I think its possible the subprocess just lives on forever